plaso/parsers/aws_elb_access.py

```diff
@@ -0,0 +1,312 @@
+      pyparsing.quotedString.setResultsName('user_agent') \
+      .setParseAction(pyparsing.removeQuotes) +
+      _WORD.setResultsName('ssl_cipher') +
+      _WORD.setResultsName('ssl_protocol') +
+      _WORD.setResultsName('target_group_arn') +
+      _WORD.setResultsName('trace_id') +
+      pyparsing.quotedString.setResultsName('domain_name') \
+      .setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName('chosen_cert_arn') \
+      .setParseAction(pyparsing.removeQuotes) +
+      _INTEGER.setResultsName('matched_rule_priority') +
+      _DATE_TIME_ISOFORMAT_STRING.setResultsName('request_creation_time') +
+      pyparsing.quotedString.setResultsName('actions_executed')  \
+      .setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName('redirect_url') \
+      .setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName('error_reason') \
+      .setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName('target_port_list') \
+      .setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName('target_status_code_list') \
+      .setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName('classification')  \
+      .setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName('classification_reason')  \
+      .setParseAction(pyparsing.removeQuotes)
+  )
+
+  LINE_STRUCTURES = [('elb_accesslog', _LOG_LINE)]
+
+
+  def ParseRecord(self, parser_mediator, key, structure):
+    """Parses a log record structure and produces events.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      key (str): name of the parsed structure.
+      structure (pyparsing.ParseResults): structure parsed from the log file.
+
+    Raises:
+      ParseError: when the structure type is unknown.
+    """
+
+    if key != 'elb_accesslog':
+      raise errors.ParseError(
+          'Unable to parse record, unknown structure: {0:s}'.format(key))
+
+    time_elements_structure = structure.get('time')
+    if time_elements_structure:
+      year, month, day_of_month, hours, minutes, seconds, microseconds = (
+          time_elements_structure)
+      time_elements_tuple = (year, month, day_of_month, hours, minutes,
+          seconds, microseconds)
+
+    try:
+      date_time = dfdatetime_time_elements.TimeElements(
+          time_elements_tuple = time_elements_tuple)
+    except ValueError:
+      parser_mediator.ProduceExtractionWarning(
+          'invalid date time value: {0!s}'.format(time_elements_tuple))
+      return
+
+    event_data = AWSELBEventData()
+    event_data.type = self._GetValueFromStructure(structure, 'type')
+    event_data.elb = self._GetValueFromStructure(structure, 'elb')
+    event_data.client_ip_address = self._GetValueFromStructure(structure,
```

Comment:
Break after `(`

```
    event_data.client_ip_address = self._GetValueFromStructure(
        structure, 'client_ip_port').get('client_ip_address')
```

Reply:
Also chaining function calls with `.` makes it harder to debug. Try to have 1 function call per line.

Reply:
I created a private, helper function (_GetValue) to avoid having to chain function calls.